node/overseer/src/lib.rs

```diff
@@ -302,7 +324,83 @@ struct SubsystemInstance<M> {
 #[derive(Debug)]
 pub struct OverseerSubsystemContext<M>{
 	rx: mpsc::Receiver<FromOverseer<M>>,
-	tx: mpsc::Sender<ToOverseer>,
+	tx: mpsc::Sender<MaybeTimed<ToOverseer>>,
+	metrics: Metrics,
+	capture_each: u32,
+	capture_state: u32,
+}
+
+impl<M> OverseerSubsystemContext<M> {
+	/// Create a new `OverseerSubsystemContext`.
+	///
+	/// `capture_each` determines which messages are timed, i.e. a value of 256 means that every 256th
+	/// message is timed. A value of `0` has a special meaning: no messages are timed.
```

Comment:
I'd be in favour of either encoding the parameter as en enum with a `From<usize>` implementation for the common case, where the `Disabled` state is explicit. Special meaning numbers tend to become a burden when reading code in the future.

Reply:
Resolved in ea59309: went back to the simpler `capture_rate` style, which has no special cases.